repos/system_upgrade/el7toel8/actors/checksendmail/actor.py

```diff
@@ -0,0 +1,76 @@
+from leapp.actors import Actor
+from leapp.models import InstalledRedHatSignedRPM, SendmailMigrationDecision
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.reporting import Report
+from leapp.libraries.common.reporting import report_with_remediation, report_generic
+import os
+import re
+import glob
+
+TCPWrappersFiles = ['/etc/hosts.allow', '/etc/hosts.deny']
+SendmailConfDir = '/etc/mail'
+SendmailConfFiles = ['/etc/mail/sendmail.cf', '/etc/mail/sendmail.mc', '/etc/mail/submit.cf', '/etc/mail/submit.mc']
+# false positives blacklist
+rfp = re.compile(r'(^\s*RIPv6:::1\b)|(@\s+\[IPv6:::1\]\s+>)')
+
+def check_wrappers_used(s):
+    r = re.compile(r'\s*' + s + r'\b')
+    for f in TCPWrappersFiles:
+        for l in open(f):
+            if r.match(l) is not None:
+                return True
+    return False
+
+def check_false_positives(f, l):
+    return f in ["/etc/mail/sendmail.cf", "/etc/mail/submit.cf"] and rfp.search(l) is not None
+
+def check_files_for_compressed_ipv6():
+    migrate_files = []
+    g = glob.glob(SendmailConfDir + '/*.db')
+    files = list(map(lambda x: re.sub(r'\.db$', '', x), g)) + SendmailConfFiles
+    r = re.compile(r'IPv6:[0-9a-fA-F:]*::')
+    for f in files:
+        if os.path.exists(f):
+            for l in open(f):
+                m = r.search(l)
+                if m is not None and not check_false_positives(f, l):
+                    migrate_files.append(f)
+    return migrate_files
```

Comment:
Hi, just looking here, please move all functiones above into the actor's library. Regarding the contribution guideline, we require to not put code in the actor.py outside of the main class. You can put it even into the class, but I suggest you to put it into the library, so even unit tests can be created for such functions.

Reply:
Fixed